diff -up mc-4.8.4/src/editor/editcmd.c.altminus mc-4.8.4/src/editor/editcmd.c
--- mc-4.8.4/src/editor/editcmd.c.altminus	2012-07-18 11:30:23.000000000 +0200
+++ mc-4.8.4/src/editor/editcmd.c	2012-07-31 11:21:14.261033325 +0200
@@ -3496,7 +3496,7 @@ edit_load_back_cmd (WEdit * edit)
         return FALSE;
 
     edit_stack_iterator--;
-    if (edit_history_moveto[edit_stack_iterator].filename_vpath != NULL)
+    if (edit_stack_iterator >= 0 && edit_history_moveto[edit_stack_iterator].filename_vpath != NULL)
         return edit_reload_line (edit, edit_history_moveto[edit_stack_iterator].filename_vpath,
                                  edit_history_moveto[edit_stack_iterator].line);
 
