--- mc-4.6.1/lib/mc.ext.in.extensions	2005-07-23 18:51:15.000000000 +0200
+++ mc-4.6.1/lib/mc.ext.in	2005-07-25 09:53:08.367448216 +0200
@@ -210,14 +210,25 @@ shell/.7z
 
 # C
 shell/.c
-	Open=%var{EDITOR:vi} %f
+	Include=editor
 
 # Fortran
 shell/.f
-	Open=%var{EDITOR:vi} %f
+	Include=editor
 	
 # Header
 regex/\.(h|hpp)$
+	Include=editor
+
+# Asm
+regex/\.([Ss]|[Aa]sm|ASM)$
+	Include=editor
+
+# C++
+regex/\.(C|cc|[Cc]pp|CPP)$
+	Inlude=editor
+
+include/editor
 	Open=%var{EDITOR:vi} %f
 
 # Object
@@ -242,9 +253,11 @@ regex/\.(te?xi|texinfo)$
 # GNU Info page
 type/^Info\ text
 	Open=info -f %f
+	#Open=pinfo %f
 
 shell/.info
 	Open=info -f %f
+	#Open=pinfo %f
 
 # Manual page
 # Exception - .so libraries are not manual pages
@@ -313,6 +326,12 @@ type/^PPM
 type/^Netpbm
 	Include=image
 
+type/^JNG
+	Include=image
+
+type/^MNG
+	Include=image
+
 shell/.xcf
 	Open=gimp %f
 
@@ -324,7 +343,7 @@ shell/.xpm
 	View=sxpm %f
 
 include/image
-	Open=if [ "$DISPLAY" = "" ]; then zgv %f; else (gqview %f &); fi
+	Open=if [ "$DISPLAY" = "" ]; then zgv %f; else (display %f &); fi
 	View=%view{ascii} identify %f
 	#View=%view{ascii} asciiview %f
 
@@ -342,8 +361,12 @@ regex/\.([wW][aA][wW]22)$
        Open=vplay -s 22 %f
 
 regex/\.([mM][pP]3)$
-	Open=if [ "$DISPLAY" = "" ]; then mpg123 %f; else (xmms %f &); fi
-	View=%view{ascii} mpg123 -vtn1 %f 2>&1 | sed -n '/^Title/,/^Comment/p;/^MPEG/,/^Audio/p'
+	Include=audio
+	#Open=if [ "$DISPLAY" = "" ]; then mpg123 %f; else (xmms %f &); fi
+	#View=%view{ascii} mpg123 -vtn1 %f 2>&1 | sed -n '/^Title/,/^Comment/p;/^MPEG/,/^Audio/p'
+
+regex/\.([mM][kK][aA])$
+	Include=audio
 
 regex/\.([oO][gG][gG])$
 	Open=if [ "$DISPLAY" = "" ]; then ogg123 %f; else (xmms %f &); fi
@@ -356,6 +379,9 @@ regex/\.([wW][mM][aA])$
 	Open=mplayer -vo null %f
 	View=%view{ascii} mplayer -quiet -slave -frames 0 -vo null -ao null -identify %f 2>/dev/null | tail +13 || file %f
 
+Include/audio
+	Open=mplayer %f
+	View=%view{ascii} mplayer -identify -vo null -ao null -frames 0 %f 2>&1 | sed -n '/^ID_/p'
 
 ### Play lists ###
 
@@ -380,20 +406,28 @@ regex/\.([mM][oO][vV]|[qQ][tT])$
 regex/\.([mM][pP]4|[mM][pP][eE]?[gG])$
 	Include=video
 
+regex/\.([mM][kK][vV])$
+	Include=video
+
 regex/\.([vV][oO][bB])$
 	Include=video
 
 regex/\.([wW][mM][vV])$
 	Include=video
 
+regex/\.([fF][lL][iIcC])$
+	Include=video
+
 regex/\.([oO][gG][mM])$
 	Include=video
 
 regex/\.([rR][aA]?[mM])$
 	Open=(realplay %f >/dev/null 2>&1 &)
+	View=%view{ascii} mplayer -identify -vo null -ao null -frames 0 %f 2>&1 | sed -n '/^ID_/p'
 
 include/video
 	Open=(mplayer %f >/dev/null 2>&1 &)
+	View=%view{ascii} mplayer -identify -vo null -ao null -frames 0 %f 2>&1 | sed -n '/^ID_/p'
 	#Open=(gtv %f >/dev/null 2>&1 &)
 	#Open=(xanim %f >/dev/null 2>&1 &)
 
