--- mc-4.6.1-20041103/lib/mc.ext.in.extensions	2004-10-28 08:19:23.000000000 +0200
+++ mc-4.6.1-20041103/lib/mc.ext.in	2004-11-03 13:20:09.889755600 +0100
@@ -108,14 +108,25 @@
 
 # C
 shell/.c
-	Open=%var{EDITOR:vi} %f
+	Include=editor
 
 # Fortran
 shell/.f
-	Open=%var{EDITOR:vi} %f
+	Include=editor
 	
 # Header
 regex/\.(h|hpp)$
+	Include=editor
+
+# Asm
+regex/\.([Ss]|[Aa]sm|ASM)$
+	Include=editor
+
+# C++
+regex/\.(C|cc|[Cc]pp|CPP)$
+	Inlude=editor
+
+include/editor
 	Open=%var{EDITOR:vi} %f
 
 # Object
@@ -123,14 +134,6 @@
 	#Open=%var{PAGER:more} %f
 	View=%view{ascii} file %f && nm %f
 
-# Asm
-shell/.s
-	Open=%var{EDITOR:vi} %f
-
-# C++
-regex/\.(C|cc|cpp)$
-	Open=%var{EDITOR:vi} %f
-
 ### Documentation ###
 
 # Texinfo
@@ -139,9 +142,11 @@
 # GNU Info page
 type/^Info\ text
 	Open=info -f %f
+	#Open=pinfo %f
 
 shell/.info
 	Open=info -f %f
+	#Open=pinfo %f
 
 # Manual page
 # Exception - .so libraries are not manual pages
@@ -210,6 +215,12 @@
 type/^Netpbm
 	Include=image
 
+type/^JNG
+	Include=image
+
+type/^MNG
+	Include=image
+
 shell/.xcf
 	Open=gimp %f
 
@@ -221,7 +232,7 @@
 	View=sxpm %f
 
 include/image
-	Open=if [ "$DISPLAY" = "" ]; then zgv %f; else (gqview %f &); fi
+	Open=if [ "$DISPLAY" = "" ]; then zgv %f; else (display %f &); fi
 	View=%view{ascii} identify %f
 	#View=%view{ascii} asciiview %f
 
@@ -239,8 +250,12 @@
        Open=vplay -s 22 %f
 
 regex/\.([mM][pP]3)$
-	Open=if [ "$DISPLAY" = "" ]; then mpg123 %f; else (xmms %f &); fi
-	View=%view{ascii} mpg123 -vtn1 %f 2>&1 | sed -n '/^Title/,/^Comment/p;/^MPEG/,/^Audio/p'
+	Include=audio
+	#Open=if [ "$DISPLAY" = "" ]; then mpg123 %f; else (xmms %f &); fi
+	#View=%view{ascii} mpg123 -vtn1 %f 2>&1 | sed -n '/^Title/,/^Comment/p;/^MPEG/,/^Audio/p'
+
+regex/\.([mM][kK][aA])$
+	Include=audio
 
 regex/\.([oO][gG][gG])$
 	Open=if [ "$DISPLAY" = "" ]; then ogg123 %f; else (xmms %f &); fi
@@ -253,6 +268,9 @@
 	Open=mplayer -vo null %f
 	View=%view{ascii} mplayer -quiet -slave -frames 0 -vo null -ao null -identify %f 2>/dev/null | tail +13 || file %f
 
+Include/audio
+	Open=mplayer %f
+	View=%view{ascii} mplayer -identify -vo null -ao null -frames 0 %f 2>&1 | sed -n '/^ID_/p'
 
 ### Play lists ###
 
@@ -277,20 +295,28 @@
 regex/\.([mM][pP]4|[mM][pP][eE]?[gG])$
 	Include=video
 
+regex/\.([mM][kK][vV])$
+	Include=video
+
 regex/\.([vV][oO][bB])$
 	Include=video
 
 regex/\.([wW][mM][vV])$
 	Include=video
 
+regex/\.([fF][lL][iIcC])$
+	Include=video
+
 regex/\.([oO][gG][mM])$
 	Include=video
 
 regex/\.([rR][aA]?[mM])$
 	Open=(realplay %f >/dev/null 2>&1 &)
+	View=%view{ascii} mplayer -identify -vo null -ao null -frames 0 %f 2>&1 | sed -n '/^ID_/p'
 
 include/video
 	Open=(mplayer %f >/dev/null 2>&1 &)
+	View=%view{ascii} mplayer -identify -vo null -ao null -frames 0 %f 2>&1 | sed -n '/^ID_/p'
 	#Open=(gtv %f >/dev/null 2>&1 &)
 	#Open=(xanim %f >/dev/null 2>&1 &)
 
