diff -up mc-4.7.0-pre1/configure.ac.extensions mc-4.7.0-pre1/configure.ac
--- mc-4.7.0-pre1/configure.ac.extensions	2009-07-31 18:36:24.000000000 +0200
+++ mc-4.7.0-pre1/configure.ac	2009-08-05 12:51:08.000000000 +0200
@@ -67,7 +67,7 @@ AC_PROG_LN_S
 AC_CHECK_TOOL(AR, ar, ar)
 
 dnl Only list browsers here that can be run in background (i.e. with `&')
-AC_CHECK_PROGS(X11_WWW, [gnome-moz-remote mozilla konqueror opera netscape])
+AC_CHECK_PROGS(X11_WWW, [firefox gnome-moz-remote mozilla konqueror opera netscape])
 
 dnl
 dnl Ovverriding mmap support.  This has to be before AC_FUNC_MMAP is used.
diff -up mc-4.7.0-pre1/misc/mc.ext.in.extensions mc-4.7.0-pre1/misc/mc.ext.in
--- mc-4.7.0-pre1/misc/mc.ext.in.extensions	2009-06-23 20:55:11.000000000 +0200
+++ mc-4.7.0-pre1/misc/mc.ext.in	2009-08-05 13:01:48.000000000 +0200
@@ -221,8 +221,8 @@ regex/\.rpm$
 
 # deb
 regex/\.u?deb$
-	Open=%cd %p#deb
-	View=%view{ascii} dpkg-deb -I %f && echo && dpkg-deb -c %f
+	Open=%cd %p#uar
+	View=%view{ascii} file %f && nm %f
 
 # dpkg
 shell/.debd
@@ -397,11 +397,16 @@ regex/\.([wW][aA][wW]22)$
        Open=vplay -s 22 %f
 
 regex/\.([mM][pP]3)$
-	Open=if [ "$DISPLAY" = "" ]; then mpg123 %f; else (xmms %f >/dev/null 2>&1 &); fi
-	View=%view{ascii} mpg123 -vtn1 %f 2>&1 | sed -n '/^Title/,/^Comment/p;/^MPEG/,/^Audio/p'
+	Include=audio
+#	Open=if [ "$DISPLAY" = "" ]; then mpg123 %f; else (xmms %f >/dev/null 2>&1 &); fi
+#	View=%view{ascii} mpg123 -vtn1 %f 2>&1 | sed -n '/^Title/,/^Comment/p;/^MPEG/,/^Audio/p'
+
+regex/\.([mM][kK][aA])$
+	Include=audio
 
 regex/\.([oO][gG][gG|aA|vV|xX])$
-	Open=if [ "$DISPLAY" = "" ]; then ogg123 %f; else (xmms %f >/dev/null 2>&1 &); fi
+	Open=ogg123 %f
+#	Open=if [ "$DISPLAY" = "" ]; then ogg123 %f; else (xmms %f >/dev/null 2>&1 &); fi
 	View=%view{ascii} ogginfo %s
 
 regex/\.([sS][pP][xX]|[fF][lL][aA][cC])$
@@ -414,11 +419,15 @@ regex/\.([wW][mM][aA])$
 	Open=mplayer -vo null %f
 	View=%view{ascii} mplayer -quiet -slave -frames 0 -vo null -ao null -identify %f 2>/dev/null | tail +13 || file %f
 
+include/audio
+	Open=mplayer %f
+	View=%view{ascii} mplayer -identify -vo null -ao null -frames 0 %f 2>&1 | sed -n '/^ID_/p'
 
 ### Play lists ###
 
 regex/\.([mM]3[uU]|[pP][lL][sS])$
-	Open=if [ -z "$DISPLAY" ]; then mplayer -vo null -playlist %f; else (xmms -p %f >/dev/null 2>&1 &); fi
+	Open=mplayer -vo null -playlist %f
+#	Open=if [ -z "$DISPLAY" ]; then mplayer -vo null -playlist %f; else (xmms -p %f >/dev/null 2>&1 &); fi
 
 
 ### Video ###
@@ -432,6 +441,9 @@ regex/\.([aA][sS][fFxX])$
 regex/\.([dD][iI][vV][xX])$
 	Include=video
 
+regex/\.([mM][kK][vV])$
+	Include=video
+
 regex/\.([mM][oO][vV]|[qQ][tT])$
 	Include=video
 
@@ -496,22 +508,27 @@ shell/.abw
 
 # Microsoft Word Document
 regex/\.([Dd][oO][cCtT]|[Ww][rR][iI])$
-	Open=(abiword %f >/dev/null 2>&1 &)
+	Open=(ooffice %f &)
+#	Open=(abiword %f >/dev/null 2>&1 &)
 	View=%view{ascii} antiword -t %f || catdoc -w %f || word2x -f text %f - || strings %f
 type/^Microsoft\ Word
-	Open=(abiword %f >/dev/null 2>&1 &)
+	Open=(ooffice %f &)
+#	Open=(abiword %f >/dev/null 2>&1 &)
 	View=%view{ascii} antiword -t %f || catdoc -w %f || word2x -f text %f - || strings %f
 
 # RTF document
 regex/\.([rR][tT][fF])$
-	Open=(abiword %f >/dev/null 2>&1 &)
+	Open=(ooffice %f &)
+#	Open=(abiword %f >/dev/null 2>&1 &)
 
 # Microsoft Excel Worksheet
 regex/\.([xX][lL][sSwW])$
-	Open=(gnumeric %f >/dev/null 2>&1 &)
+	Open=(ooffice %f &)
+#	Open=(gnumeric %f >/dev/null 2>&1 &)
 	View=%view{ascii} xls2csv %f || strings %f
 type/^Microsoft\ Excel
-	Open=(gnumeric %f >/dev/null 2>&1 &)
+	Open=(ooffice %f &)
+#	Open=(gnumeric %f >/dev/null 2>&1 &)
 	View=%view{ascii} xls2csv %f || strings %f
 
 # Use OpenOffice.org to open any MS Office documents
