diff -up mc-4.7.5-pre1/src/filemanager/filegui.c.fileguicrash mc-4.7.5-pre1/src/filemanager/filegui.c
--- mc-4.7.5-pre1/src/filemanager/filegui.c.fileguicrash	2010-12-06 12:03:56.000000000 +0100
+++ mc-4.7.5-pre1/src/filemanager/filegui.c	2010-12-09 08:12:33.452156885 +0100
@@ -617,7 +617,7 @@ file_op_context_create_ui_without_init (
     if ((right_panel == current_panel) && !classic_progressbar)
     {
         ui->progress_file_gauge->from_left_to_right = FALSE;
-        if (dialog_type == FILEGUI_DIALOG_MULTI_ITEM)
+        if (verbose && dialog_type == FILEGUI_DIALOG_MULTI_ITEM)
             ui->progress_total_gauge->from_left_to_right = FALSE;
     }
 }
