diff -up mc-4.7.4/lib/search/glob.c.globfix mc-4.7.4/lib/search/glob.c
--- mc-4.7.4/lib/search/glob.c.globfix	2010-09-07 09:16:43.000000000 +0200
+++ mc-4.7.4/lib/search/glob.c	2010-10-07 08:43:56.517785647 +0200
@@ -144,7 +144,6 @@ mc_search__translate_replace_glob_to_reg
             {
                 g_string_append_c (buff, '\\');
                 c = ++cnt;
-                continue;
             }
             break;
             /* breaks copying: mc uses "\0" internally, it must not be changed */
