diff -up mc-4.7.2.99/src/editor/editwidget.c.mcviewsegfault mc-4.7.2.99/src/editor/editwidget.c
--- mc-4.7.2.99/src/editor/editwidget.c.mcviewsegfault	2010-06-08 13:23:03.000000000 +0200
+++ mc-4.7.2.99/src/editor/editwidget.c	2010-06-08 13:28:14.960729778 +0200
@@ -286,7 +286,7 @@ edit_file (const char *_file, int line)
 const char *
 edit_get_file_name (const WEdit * edit)
 {
-    return edit->filename;
+    return edit?edit->filename:NULL;
 }
 
 void
