diff -up mc-4.7.5-pre1/lib/vfs/mc-vfs/direntry.c.vfscrash mc-4.7.5-pre1/lib/vfs/mc-vfs/direntry.c
--- mc-4.7.5-pre1/lib/vfs/mc-vfs/direntry.c.vfscrash	2010-12-06 12:03:44.000000000 +0100
+++ mc-4.7.5-pre1/lib/vfs/mc-vfs/direntry.c	2010-12-08 14:27:08.975092483 +0100
@@ -1166,8 +1166,12 @@ vfs_s_open (struct vfs_class *me, const 
         }
 
         split_dir_name (me, q, &dirname, &name, &save);
-        /* FIXME: check if vfs_s_find_inode returns NULL */
         dir = vfs_s_find_inode (me, super, dirname, LINK_FOLLOW, FL_DIR);
+        if (dir == NULL)
+        {
+            g_free (q);
+            return NULL;
+        }
         if (save)
             *save = PATH_SEP;
         ent = vfs_s_generate_entry (me, name, dir, 0755);
diff -up mc-4.7.5-pre1/lib/vfs/mc-vfs/gc.c.vfscrash mc-4.7.5-pre1/lib/vfs/mc-vfs/gc.c
--- mc-4.7.5-pre1/lib/vfs/mc-vfs/gc.c.vfscrash	2010-12-03 09:53:38.000000000 +0100
+++ mc-4.7.5-pre1/lib/vfs/mc-vfs/gc.c	2010-12-08 14:35:37.886866014 +0100
@@ -222,7 +222,7 @@ vfs_stamp_create (struct vfs_class *oldv
         n3vfsid = NULL;
     }
 
-    if (!oldvfs->nothingisopen || !(*oldvfs->nothingisopen) (oldvfsid))
+    if (!oldvfs || !oldvfs->nothingisopen || !(*oldvfs->nothingisopen) (oldvfsid))
         return;
 
     vfs_addstamp (oldvfs, oldvfsid);
